pkg/cmd/server/start/node.go

```diff
@@ -0,0 +1,69 @@
+package start
+
+import (
+	_ "net/http/pprof"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	"github.com/openshift/origin/pkg/cmd/server/kubernetes"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/util/docker"
+)
+
+func StartNode(config configapi.NodeConfig) error {
+	if config.RecordEvents {
+		kubeClient, err := GetNodeKubeClient(config)
+		if err != nil {
+			return err
+		}
+
+		// TODO: recording should occur in individual components
+		record.StartRecording(kubeClient.Events(""), kapi.EventSource{Component: "node"})
+	}
+
+	nodeConfig, err := BuildKubernetesNodeConfig(config)
+	if err != nil {
+		return err
+	}
+
+	nodeConfig.EnsureVolumeDir()
+	nodeConfig.EnsureDocker(docker.NewHelper())
+	nodeConfig.RunProxy()
+	nodeConfig.RunKubelet()
+
+	return nil
+}
+
+func BuildKubernetesNodeConfig(originalConfig configapi.NodeConfig) (*kubernetes.NodeConfig, error) {
+	kubeClient, err := GetNodeKubeClient(originalConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &kubernetes.NodeConfig{
+		NodeHost: originalConfig.NodeID,
```

Comment:
It's super confusing for kubernetes.NodeConfig.NodeHost to not come from configapi.NodeConfig.NodeHost